### DISCUSSION TITLE
How to make the most of file.last_modified_date

### DISCUSSION THREAD

**Discussion by thor314** [SilentVoid13/Templater#912] [Help]
Per https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilelast_modified_dateformat-string--yyyy-mm-dd-hhmm

When I use the function in a template, it will only ever be evaluated once, not recurringly updated every time I poke a file. This reduces the utility for templates to basically equivalent to file_creation_date. 
Is there a convenient workaround for this?

**Comment by Zachatoo** [comment]
I would recommend using Dataview instead. You _can_ use Templater using dynamic commands, but I don't recommend using those, they don't always work and will likely be deprecated since Dataview can do pretty much everything dynamic commands can do and more.
https://blacksmithgu.github.io/obsidian-dataview/queries/dql-js-inline#inline-dql

Code would look something like this:
```js
`= this.file.mtime`
```

**Reply by thor314** [reply]
@Zachatoo, yeah I stumbled across this snippet after looking into it for awhile.

 I suppose the file already knows its last modified date, but the metadata field is convenient for human readability, even if the file already knows.
 I've taken to putting this line at the top of each file template, just below the frontmatter, since it would throw an error in the frontmatter.
*last-modified: `$= dv.current().file.mtime`*
